plotloops!

danielduyvo opened the pull request:
* Added plotloops! function
* Added documentation for the plotloops!
* Added tests for plotloops!
* Added an example figure for plotloops! to the README.md
Unrelated but accidentally snuck in
* Updated compatibility for CairoMakie and Makie to the latest version

danielduyvo committed:
Update compatibility for Makie

danielduyvo committed:
Add plotloops! function

danielduyvo committed:
Add plotloops! figure to README

Only included the basic loops figure and deleted the colored and
overlayed versions; did not feel like it was displaying enough added
functionality to warrant the space taken.

mmkim1210 commented:
@danielduyvo looks good! could you change the font to arial for the figures  so that the aesthetics is constant throughout the documentation?

danielduyvo committed:
Draw arcs as lines, with paired ends as rectangles

The ellipseband! function was removed and replaced with drawloop!
function, which draws a single line (instead of a thick arc) from the
midpoint of each paired end. The paired ends have rectangles to indicate
the width of each end. The plotloops! function's keyword arguments have
been updated accordingly. Additionally, the length and step keywords for
plotloops! have been removed and replaced with resolution, which will
take in a number of points to draw across the provided range on the
x-axis. The loops will be drawn to a consistent resolution by that
keyword argument. Tests and documentation have been updated, and the
example figure in the README has been removed.

danielduyvo committed:
Set font to Arial in plotloops! figures
